perfkitbenchmarker/rackspace/rackspace_virtual_machine.py

```diff
@@ -0,0 +1,434 @@
+      self.ip_address = instance['accessIPv4']
+      self.ip_address6 = instance['accessIPv6']
+      self.internal_ip = instance['private network']
+    else:
+      raise errors.Error('Unexpected failure.')
+
+  def _Delete(self):
+    """Delete a Rackspace VM instance."""
+    env = os.environ.copy()
+    env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    delete_cmd = [FLAGS.nova_path, 'delete', self.id]
+    vm_util.IssueCommand(delete_cmd, env=env)
+
+  def _Exists(self):
+    """Returns true if the VM exists."""
+    env = os.environ.copy()
+    env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    getinstance_cmd = [FLAGS.nova_path, 'show', self.id]
+    stdout, stderr, _ = vm_util.IssueCommand(getinstance_cmd, env=env)
+    if stdout.strip() == '':
+        return False
+    instance = util.ParseNovaTable(stdout)
+    return (instance.get('OS-EXT-STS:task_state') == 'deleting' or
+            instance.get('status') == 'ACTIVE')
+
+  def _GetFlavorDetails(self):
+    """Retrieves details about the flavor used to build the instance.
+
+    Returns:
+      A dict of properties of the flavor used to build the instance.
+    """
+    env = os.environ.copy()
+    env.update(util.GetDefaultRackspaceNovaEnv(self.zone))
+    flavor_show_cmd = [FLAGS.nova_path, 'flavor-show', self.machine_type]
+    stdout, _, _ = vm_util.IssueCommand(flavor_show_cmd, env=env)
+    flavor_properties = util.ParseNovaTable(stdout)
+    return flavor_properties
+
+  def _GetBlockDevices(self):
+    stdout, _ = self.RemoteCommand(
+        'sudo lsblk -o NAME,MODEL,SIZE,TYPE,MOUNTPOINT,LABEL -n -b -P')
+    lines = stdout.splitlines()
+    groups = [LSBLK_PATTERN.match(line) for line in lines]
+    tuples = [g.groups() for g in groups if g]
+    colnames = ('name', 'model', 'size_bytes', 'type', 'mountpoint', 'label',)
+    blk_devices = [dict(zip(colnames, t)) for t in tuples]
+    for d in blk_devices:
+      d['model'] = d['model'].rstrip()
+      d['label'] = d['label'].rstrip()
+      d['size_bytes'] = int(d['size_bytes'])
+    return blk_devices
+
+  def _GetBootDevice(self):
+    blk_devices = self._GetBlockDevices()
+    boot_blk_device = None
+    for dev in blk_devices:
+      if dev['mountpoint'] == '/':
+        boot_blk_device = dev
+        break
+
+    if boot_blk_device is None:  # Unlikely
+      raise errors.Error('Could not find disk with "/" root mount point.')
+
+    if boot_blk_device['type'] == 'part':
+      blk_device_name = boot_blk_device['name'].rstrip('0123456789')
+      for dev in blk_devices:
+        if dev['type'] == 'disk' and dev['name'] == blk_device_name:
+          boot_blk_device = dev
+          break
+      else:  # Also, unlikely
+        raise errors.Error('Could not find disk containing boot partition.')
+
+    return boot_blk_device
+
+  def _ApplyOnMetalDiskTuning(self, disks):
+    """Applies recommended I/O scheduler and queue tuning for OnMetal IO
```

Comment:
@klausw First of all, thank you for taking the time to review this. :)

I'm okay removing it, and leaving them optional with a flag, as you suggested. But as for the tuning used in OnMetal IO machine type, it is a combination of the recommend tuning from the flash drives manufacturer, and our own. This tuning wasn't added to the images to keep them as vanilla as possible, but otherwise, it would have been included with them. Using the OS default settings for these drives, will result in sub-optimal performance.

This documented here in our Rackspace Support docs: http://www.rackspace.com/knowledge_center/article/configure-flash-drives-in-high-io-instances-as-data-drives